packages/xo-web/src/xo-app/vm/tab-network.js

```diff
@@ -527,12 +538,23 @@ export default class TabNetwork extends BaseComponent {
       newVif: !this.state.newVif,
     })
 
-  _getIpAddresses = createSelector(
+  _getIpsByDevice = createSelector(
     () => this.props.vm.addresses,
-    // VM_guest_metrics.networks seems to always have 3 fields (ip, ipv4 and ipv6) for each interface
-    // http://xenbits.xenproject.org/docs/4.12-testing/misc/xenstore-paths.html#attrvifdevidipv4index-ipv4_address-w
-    // https://github.com/xapi-project/xen-api/blob/d650621ba7b64a82aeb77deca787acb059636eaf/ocaml/xapi/xapi_guest_agent.ml#L76-L79
-    addresses => uniq(values(addresses))
+    addresses => {
+      // VM_guest_metrics.networks seems to always have 3 fields (ip, ipv4 and ipv6) for each interface
+      // http://xenbits.xenproject.org/docs/4.12-testing/misc/xenstore-paths.html#attrvifdevidipv4index-ipv4_address-w
+      // https://github.com/xapi-project/xen-api/blob/d650621ba7b64a82aeb77deca787acb059636eaf/ocaml/xapi/xapi_guest_agent.ml#L76-L79
+      const ipsByDevice = {}
+      Object.keys(addresses).forEach(key => {
```

Comment:
You can use `Object.entries` if you know you'll need both the key and the value.